### PR TITLE
feat: pre-mining data and contract call 

### PR DESCRIPTION
**Pull request by keeplearning-near**

### CODE REVIEW

File: src/components/menu/Aidrdrop/AirdropModal.tsx [hunk 1]
```diff
@@ -11,6 +11,48 @@ import { useUserWalletDataContext } from '../../../libs/web3-data-provider';
 import { isValid } from '../../../helpers/number';
 import { useProtocolDataContext } from '../../../libs/protocol-data-provider';
 
+const UIINCENTIVEDATAPROVIDER_ABI = [
```
Comment (linguists) [inline src/components/menu/Aidrdrop/AirdropModal.tsx:14]: We should probably query data via `@pret/contract-helper` package. 

We need to update the package here: https://github.com/pret-labs/pret-utilities/blob/master/packages/contract-helpers/src/ui-incentive-data-provider/index.ts

File: src/ui-config/networks.ts [hunk 2]
```diff
@@ -260,6 +267,7 @@ export const networkConfigs: Record<string, BaseNetworkConfig> = {
       url: 'https://rainbowbridge.app/',
       logo: auroraBridgeLogo,
     },
+    cornIncentivesController: '0xe303206ac6065ae2a53553116da7ddb54fd24776',
```
Comment (linguists) [inline src/ui-config/networks.ts:270]: This is duplicate with the 2nd `incentiveDataControllers`

File: src/ui-config/markets/index.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ export const marketsData: { [key in keyof typeof CustomMarket]: MarketDataType }
       incentives: true,
     },
     addresses: {
-      LENDING_POOL_ADDRESS_PROVIDER: '0x8fcc9f65BF153371f4B06542446eE51aCcDb635C'.toLowerCase(),
+      LENDING_POOL_ADDRESS_PROVIDER: '0x94de7fcd50f33d2598639f9abafed90ba0eb9a11'.toLowerCase(),
       LENDING_POOL: '0x8ca1db9e43305636e9b4bf8ca1ff8f97579009ad',
```
Comment (linguists) [inline src/ui-config/markets/index.ts:34]: `LENDING_POOL` and `WETH_GATEWAY` also need to be updated.

**Review by linguists — CHANGES_REQUESTED** [review]

**Comment by zhubajie8881** [comment]
test passed

File: src/ui-config/networks.ts [hunk 4]
```diff
@@ -260,6 +271,7 @@ export const networkConfigs: Record<string, BaseNetworkConfig> = {
       url: 'https://rainbowbridge.app/',
       logo: auroraBridgeLogo,
     },
+    cornIncentivesController: '0xE303206ac6065aE2a53553116da7ddb54fd24776',
```
Comment (linguists) [inline src/ui-config/networks.ts:274]: Should we add this into `addresses.incentiveControllers`?

File: src/components/menu/Aidrdrop/AirdropModal.tsx [hunk 5]
```diff
@@ -155,25 +161,46 @@ function SpaceLine({ style }: { style?: CSSProperties }) {
...
 
+  const { library: provider } = useWeb3React<providers.Web3Provider>();
+
   const { currentAccount } = useUserWalletDataContext();
-  const [contract, setContract] = useState<ethers.Contract | null>(null);
-  const [claimableAmountRaw, setClaimableAmountRaw] = useState('0');
-  const [data, setData] = useState<{
+  const [cornAirdropContract, setCornAirdropContract] = useState<ethers.Contract | null>(null);
+  const [airdropClaimableAmountRaw, setAirdropClaimableAmountRaw] = useState('0');
+  const [preminingClaimableAmountRaw, setPreminingClaimableAmountRaw] = useState('0');
```
Comment (linguists) [inline src/components/menu/Aidrdrop/AirdropModal.tsx:174]:
```suggestion
  const [preMiningClaimableAmountRaw, setPreMiningClaimableAmountRaw] = useState('0');
```

File: src/components/menu/Aidrdrop/AirdropModal.tsx [hunk 6]
```diff
@@ -155,25 +161,46 @@ function SpaceLine({ style }: { style?: CSSProperties }) {
...
+  const [airdropData, setAirdropData] = useState<{
+    totalAmount: string;
+    pendingAmount: string;
+    claimableAmount: string;
+  }>({
+    totalAmount: '-',
+    pendingAmount: '-',
+    claimableAmount: '-',
+  });
+  const [preminingData, setPreminingData] = useState<{
```
Comment (linguists) [inline src/components/menu/Aidrdrop/AirdropModal.tsx:185]:
```suggestion
  const [preMiningData, setPreMiningData] = useState<{
```

**Review by linguists — CHANGES_REQUESTED** [review]

File: src/ui-config/networks.ts [hunk 7]
```diff
@@ -34,7 +34,11 @@ export const networkConfigs: Record<string, BaseNetworkConfig> = {
   },
   [ChainId.mainnet]: {
     name: 'Ethereum mainnet',
-    publicJsonRPCUrl: ['https://cloudflare-eth.com', 'https://eth-mainnet.alchemyapi.io/v2/demo'],
+    publicJsonRPCUrl: [
+      'https://aurora-mainnet.infura.io/v3/eee0ad1bd3c5487585685be6b7f4ef94',
```
Comment (linguists) [inline src/ui-config/networks.ts:38]: Is this RPC URL necessary?

**Review by linguists — APPROVED** [review]

**Pull request merged**